R/convert_na_to.R

```diff
@@ -169,60 +169,34 @@ convert_na_to.data.frame <- function(x,
     verbose = verbose
   )
 
-  # list are not covered by .select_nse
-  if (length(select_nse) == 0 && is.list(select)) {
-    if (inherits(exclude, "formula")) {
-      exclude <- all.vars(exclude)
+  # default
+  lookup <- lapply(x, function(y) {
+    if (is.numeric(y)) {
+      replace_num
+    } else if (is.character(y)) {
+      replace_char
+    } else if (is.factor(y)) {
+      replace_fac
```

Comment:
Not sure how making a named function would simplify the code, this function is very specific to `convert_na_to()` and is used only once in its code.

I also don't know how to use `switch()` in this context since it's a different function called at each `if` statement